service/history/historyEngine.go

```diff
@@ -974,7 +974,7 @@ func (e *historyEngineImpl) getMutableStateOrPolling(
 
 	// expectedNextEventID is 0 when caller want to get the current next event ID without blocking
 	expectedNextEventID := common.FirstEventID
-	if request.ExpectedNextEventID != nil {
+	if request.ExpectedNextEventID != 0 {
```

Comment:
I suppose a 0-id would be fairly meaningless, as you can't wait for the first (i.e. "start workflow") event?  so all practical ones are 1+... if so, this seems fine (and matches the comment)